lang/java/avro/src/main/java/org/apache/avro/reflect/ReflectData.java

```diff
@@ -337,6 +350,33 @@ static Class getClassProp(Schema schema, String prop) {
     ARRAY_CLASSES.put(boolean.class, boolean[].class);
   }
 
+  /**
+   * It returns false for non-string-maps because Avro writes out such maps
+   * as an array of records. Even their JSON representation is an array.
+   */
+  protected boolean isMap(Object datum) {
```

Comment:
Should this be `isAvroMap`? There are obvious cases where this returns false though the type is `Map`.